### PR TITLE
add circle.yml to build docker containers

### PR DESCRIPTION
**Pull request by mostlygeek**
This circle.yml replicates the asset building container and tests in the travis build.

### CODE REVIEW

**Comment by mostlygeek** [comment]
We should probably use this circle.yml instead of the basic one in #412. However, it seems the static asset build is broken. We'll need to fix that before this would pass. 

Also a note about circleci, it doesn't seem to build the PR from a fork unless that fork is also configured to be built.

**Comment by mostlygeek** [comment]
@lmorchard I think I need your help with this one. The circleci build seems to be failing building the static assets. 

I ran this manually and the same error. It seems to fail on the lint tests with: 

```
[08:27:46] Starting 'lint'...

TypeError: Cannot read property 'visitClass' of undefined
....
```

You can check out the log https://circleci.com/gh/mostlygeek/idea-town/2.

**Comment by lmorchard** [comment]
I see it, but have no idea what it means & can't reproduce in my local frontend build container. Will try poking around and see what might cause that.

**Comment by mostlygeek** [comment]
I got rid of the build problem and got flake8 working. However I seem to be failing on the `coverage run manage.py test -v2`. It seems to not be able to find the tests in the container. Did I miss a dependency?

Check out the error I'm getting here: https://circleci.com/gh/mostlygeek/idea-town/16

**Comment by lmorchard** [comment]
Got this [passing](https://circleci.com/gh/lmorchard/testpilot/12) on my Circle CI account. Had to make [a couple of tweaks](https://github.com/lmorchard/testpilot/commit/b76d3a100cae970eb337bbf279793ddae89a3326) on my branch:

``` patch
diff --git a/.coveragerc b/.coveragerc
deleted file mode 100644
index 922d953..0000000
--- a/.coveragerc
+++ /dev/null
@@ -1,5 +0,0 @@
-[run]
-source = idea_town
-
-[report]
-omit = */migrations/*
diff --git a/circle.yml b/circle.yml
index 6b63951..df889fe 100644
--- a/circle.yml
+++ b/circle.yml
@@ -41,8 +41,7 @@ dependencies:

 test:
   override:
-    - docker run -it --rm app:build flake8 idea_town
-    - docker run -it --rm --net=host -e 'DEBUG=False' -e 'SECRET_KEY=Foo' -e 'ALLOWED_HOSTS=localhost' -e 'DATABASE_URL=postgres://ubuntu@localhost/circle_test' --user root app:build sh -c 'pip install coverage && coverage run manage.py test -v2'
+    - docker run --net=host -e 'DEBUG=False' -e 'SECRET_KEY=Foo' -e 'ALLOWED_HOSTS=localhost' -e 'DATABASE_URL=postgres://ubuntu@localhost/circle_test' --user root app:build sh -c 'flake8 testpilot && ./manage.py test -v2 testpilot'

 # appropriately tag and push the container to dockerhub
 deployment:
```

Removed the coverage invocation, because we're not actually using that data yet. Also added an explicit `testpilot` module to `./manage.py test` - not sure why that made a difference since we haven't needed it on travis or local dev, but it seemed to be needed here.

**Comment by lmorchard** [comment]
Also realized this is missing linting & tests for frontend & addon that we run on Travis. If the goal is to have a fully tested container, we should probably add those too and just switch entirely over from Travis

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    # build the actual deployment container
+    - docker build -t app:build .
+
+    # Clean up any old images and save the new one
+    - I="image-$(date +%j).tar"; mkdir -p ~/docker; rm ~/docker/*; docker save app:build > ~/docker/$I; ls -l ~/docker
+
+test:
+  override:
+    - docker run -it --rm app:build flake8 idea_town
```
Comment (pdehaan) [inline circle.yml]: "idea_town"?

**Pull request closed without merging**